HHS/Head-Start-TTADP
# [TTAHUB-2139/2121] Share my groups

## Description of change

In this PR we add both the FE and BE for share my groups. This change includes new sections for groups on the 'Account Management' page. As well as the ability to share groups with co-owners (editors) and individuals on the edit group page. We also update the view group page to show creator, recipeints, coowners, and individuals.

![image](https://github.com/HHS/Head-Start-TTADP/assets/84350609/abbd7d7e-0ef2-4d23-aa47-9af6ee3a2e4e)

![image](https://github.com/HHS/Head-Start-TTADP/assets/84350609/70f211c6-38d1-4beb-b80a-9b6c4937cce3)

![image](https://github.com/HHS/Head-Start-TTADP/assets/84350609/10a737bb-59c4-4643-a7a6-8029d6e15430)

## How to test

- Review the code
- Create a group that is shared with both co-owners and individuals.
- VERIFY: All users in the co-owner and indvidual list have access to (read, write, approve) reports in the exact same regions.
- Make sure you can make updates to that group.
- Impersonate both the co-owner and individuals you should see that group in the appropriate section.
- Make sure as a co-owner you can edit and as an individual you can only view.
- Make sure as both roles you can assign the group to an AR (respectfully).
- Make sure you can still create a private group that is NOT shared with anyone.
- View a group and make sure the readonly view shows the correct values.

**Note:** If you create a group with region 2 grants and your user defaults to creating grants in region 1 you will NOT see the group.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2139


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
